### ISSUE TITLE
Remove `divisible` keyword

### ISSUE THREAD

**Issue by libscott** [bigchaindb/bigchaindb#841]
It seems that a `divisible`, non `updatable`, non `refillable` asset with an amount `1` would be equal to a non-divisible asset, so perhaps we can remove the `divisible` keyword. I think it would be worthwhile as it would remove a bit of complexity from tests and make our product less conceptually complicated.

**Comment by libscott** [comment]
Have a branch for this: https://github.com/bigchaindb/bigchaindb/tree/remove-divisible-keyword.

**Comment by ttmc** [comment]
Oh interesting. So you're saying that the amount already determines whether an asset is divisible or not?

What about a `divisible` asset that is `refillable` and starts out with an amount `1`? Initially, it will be effectively indivisible, but later, if it gets more amount added to it, its divisibility would become meaningful. It's true that `refillable` assets aren't implemented today, but they may be in the future. Were you thinking that the `divisible` keyword/flag could be added back at that time?

P.S. I took a look at the `remove-divisible-keyword` branch and see that you spotted some places where the docs became out-of-date when divisible assets became supported. Those docs _should_ be updated, so if this idea gets shelved, let's try to remember to fix those docs at least.

**Comment by libscott** [comment]
> What about a divisible asset that is refillable and starts out with an amount 1? Initially, it will be effectively indivisible, but later, if it gets more amount added to it, its divisibility would become meaningful. It's true that refillable assets aren't implemented today, but they may be in the future. Were you thinking that the divisible keyword/flag could be added back at that time?

If it starts out with amount 1, no-one will be able to divide it, and when it has amount >= 1, it will become `divisible` automatically. In the case of `refillable` assets, they don't appear to have any bearing or whether or not we need the `divisible` keyword. The only case that we would need a `divisible` keyword is if we want to support non-divisible assets with an amount > 1, but from what I can see this is currently not the intention:

```python
    # If the amount is supplied we can perform extra validations to 
    # the asset                                                     
    if amount is not None:                                          
        if not isinstance(amount, int):                             
            raise TypeError('`amount` must be an int')              
                                                                    
        if self.divisible is False and amount != 1:                 
            raise AmountError('non divisible assets always have'    
                              ' amount equal to one')               
                                                                    
        # Since refillable assets are not yet implemented this shoul
        # raise and exception                                       
        if self.divisible is True and amount < 2:                   
            raise AmountError('divisible assets must have an amount'
                              ' greater than one')                  
```

**Comment by TimDaub** [comment]
Relevant: 
- https://github.com/bigchaindb/bigchaindb/issues/125#issuecomment-240442324
- https://github.com/bigchaindb/bigchaindb/issues/125#issuecomment-240656981

**Comment by sohkai** [comment]
Fixed with #962.
